docs(stripe): adding docs for stripe (#863)
front-commerce/developers.front-commerce.com: docs/03-extensions/stripe/index.mdx

---
title: "Stripe"
description:

>   "This extension allow the usage of Stripe in your Front-Commerce application."

Reviewer: ```suggestion
  "This extension allows the usage of Stripe in your Front-Commerce application."
```

Author: fixed

---

<SinceVersion tag="3.5" />

<p>{frontMatter.description}</p>

## Installation

First ensure you have installed the package:

```bash
$ pnpm install @front-commerce/stripe
```

## Front-Commerce configuration

[After installing the Stripe package](/docs/3.x/extensions/stripe/#installation),
you need to enable the corresponding extension. For that, you need to tweak your

> `front-commerce.config.ts` file like:

Reviewer 2: ```suggestion
`front-commerce.config.ts` file like this:
```

Author: fixed

```typescript title="front-commerce.config.ts"
import { defineConfig } from "@front-commerce/core/config";
import themeChocolatine from "@front-commerce/theme-chocolatine";
import magento2 from "@front-commerce/magento2";
// highlight-next-line
import stripe from "@front-commerce/stripe";
import storesConfig from "./app/config/stores";
import cacheConfig from "./app/config/caching";

export default defineConfig({
  extensions: [
    magento2({ storesConfig }),

>     themeChocolatine(),

Reviewer: // ⚠️ needs to be before magento2() 😇 see https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/blob/b9840a77f327f56d562fea807357d62f6243939c/skeleton/front-commerce.config.ts#L16-21

Author: fixed

    // highlight-next-line

>     stripe(), // ⚠️ need to be after themeChocolatine()

Reviewer 2: ```suggestion
    stripe(), // ⚠️ needs to be after themeChocolatine()
```

Author: fixed

  ],
  stores: storesConfig,
  cache: cacheConfig,
  configuration: {
    providers: [],
  },
});
```

## Register your Stripe payment components

:::note

> This is a temporary way to setup payment components. We are aware that it is
> tedious. It will definitely change in the future, when Front-Commerce will
> support new extension points for Payments.

Reviewer: ```suggestion
The current method for configuring payment components is an interim solution. We recognize that it may be cumbersome for users. Please be assured that this process will undergo significant improvements in the future, as Front-Commerce evolves to include more versatile extension points for Payment systems.
```

Reviewer: #gpt

:::

> 1. Override the file that lets you register additional payments forms in

Reviewer 2: ```suggestion
1. Override the file that lets you register additional payments methods in
```

Author: fixed

import storesConfig from "./app/config/stores";
import cacheConfig from "./app/config/caching";

export default defineConfig({
  extensions: [
    themeChocolatine(),
    magento2({ storesConfig }),
    // highlight-next-line
    stripe(), // ⚠️ needs to be after themeChocolatine()
  ],
  stores: storesConfig,
  cache: cacheConfig,
  configuration: {
    providers: [],
  },
});
```

## Register your Stripe payment components

:::note

The current method for configuring payment components is an interim solution. We
recognize that it may be cumbersome for users. Please be assured that this
process will undergo significant improvements in the future, as Front-Commerce
evolves to include more versatile extension points for Payment systems.

:::

1. Override the file that lets you register additional payments methods in
   Front-Commerce

   ```shell
   mkdir -p app/theme/modules/Checkout/Payment/AdditionalPaymentInformation/
   cp -u node_modules/@front-commerce/theme-chocolatine/theme/modules/Checkout/Payment/AdditionalPaymentInformation/getAdditionalDataComponent.js app/theme/modules/Checkout/Payment/AdditionalPaymentInformation/getAdditionalDataComponent.js
   ```

> 2. Register Payzen

Reviewer 2: ```suggestion
2. Register Stripe
```

Author: fixed
